src/main.rs

```diff
@@ -94,3 +70,62 @@ fn window_conf() -> Conf {
         ..Default::default()
     }
 }
+
+/// Callable logic or draw data to be run on the game loop
+trait Call {
+    /// Run data/logic that doesn't require modifying the related entity. Used for rendering.
+    fn call(&self) {}
+
+    /// Run entity logic & interaction, such as components or other game information.
+    fn call_mut(&mut self) {}
+}
+
+impl<T: ?Sized + Call> Call for &'_ T {
+    fn call(&self) {
+        <T as Call>::call(&**self);
+    }
+}
+
+impl<T: ?Sized + Call> Call for &'_ mut T {
+    fn call_mut(&mut self) {
+        <T as Call>::call_mut(&mut **self);
+    }
+    fn call(&self) {
+        <T as Call>::call(&**self);
+    }
+}
+
+/// Container for call objects.
+/// Bundle and call all gameObjects at once
+struct Scene<'s> {
+    pub function_stack: Vec<Box<dyn Call + 's>>, // Used to run/handle game logic
+}
+
+impl<'s> Scene<'s> {
+    /// Constructor.
+    /// Initializes with empty function stack
+    fn init() -> Self {
+        Scene {
+            function_stack: vec![],
+        }
+    }
+    /// Call all function logic
+    fn tick(&mut self) {
+        for function in &mut self.function_stack {
+            function.call_mut();
+            function.call();
+        }
+    }
+
+    /// Push value as mutable reference
+    /// For player, maps & other gameObjects
+    fn push_mut<T: Call>(&mut self, item: &'s mut T) {
+        self.function_stack.push(Box::new(item));
+    }
+
```

Comment:
Looking good so far but we need to make the drawstack calls more ambiguous

Reply:
Will do. Function pointer passing is a thing now with more recent commits on the branch, if we can, I think updating/making a new pull request will probably be in order soon. Not immediately though, there is still some work to be done